tools/python/paladin_claiming/recycle_bribes.py

```diff
@@ -0,0 +1,195 @@
+
+
+def fetch_claimable_paladin_bribes(address: str) -> List[dict]:
+    response = requests.get(
+        f"https://api.paladin.vote/quest/v3/copilot/claims/{address}"
+    )
+    response.raise_for_status()
+    return response.json()["claims"]
+
+
+def get_prop_hash(target: str) -> str:
+    prop = Web3.solidity_keccak(["address"], [Web3.to_checksum_address(target)])
+    return f"0x{prop.hex().lstrip('0x')}"
+
+
+def get_chain_dirs(base_dir: Path, chain_name: str) -> Path:
+    chain_dir = base_dir / chain_name
+    chain_dir.mkdir(parents=True, exist_ok=True)
+    return chain_dir
+
+
+def claim_paladin_bribes(claims: List[dict], chain_name: str) -> str:
+    """
+    - generate tx to claim all bribes from paladin
+    - save claim results to csv
+    """
+    chain_addrs = CHAIN_ADDRS[chain_name]
+    paladin_distributor = SafeContract(
+        chain_addrs["paladin_distributor"], paladin_distributor_abi
+    )
+    w3 = w3_by_chain[chain_name]
+
+    chain_dir = get_chain_dirs(CLAIM_OUTPUT_DIR, chain_name)
+    csv_path = chain_dir / f"paladin_claims_{chain_name}_{CURRENT_DATE}.csv"
+
+    with open(csv_path, "w") as f:
+        f.write("chain,token_address,gauge_address,amount,amount_mantissa\n")
+
+        for claim in claims:
+            print(f"Claiming - token: {claim['token']}, gauge: {claim['gauge']}")
+
+            paladin_distributor.claim(
+                claim["questId"],
+                claim["period"],
+                claim["index"],
+                claim["user"],
+                claim["amount"],
+                claim["proofs"],
+            )
+
+            token_contract = w3.eth.contract(
+                address=Web3.to_checksum_address(claim["token"]), abi=erc20_abi
+            )
+            decimals = token_contract.functions.decimals().call()
+            amount = int(claim["amount"]) / (10**decimals)
+
+            f.write(
+                f"{chain_name},{claim['token']},{claim['gauge']},{amount},{claim['amount']}\n"
+            )
+
+    return csv_path
+
+
+def deposit_hh_bribes(csv_path: str, chain_name: str, builder: SafeTxBuilder):
+    """
+    - generate tx to deposit 70% of claimed bribe amounts on HH
+    - save remaining 30% (DAO fee) to separate csv
+    - output final transaction payload for claims and deposits
+    """
+    chain_addrs = CHAIN_ADDRS[chain_name]
+
+    chain_dir = get_chain_dirs(CLAIM_OUTPUT_DIR, chain_name)
+    sell_csv_path = (
+        chain_dir / f"paladin_tokens_to_sell_{chain_name}_{CURRENT_DATE}.csv"
+    )
+    with open(sell_csv_path, "w") as f:
+        f.write("chain,token_address,amount,amount_mantissa\n")
+
+    # Aggregate claimed bribes by amounts by token and gauge
+    aggregated_bribes = {}
+    with open(csv_path, "r") as f:
+        next(f)
+        for line in f:
+            _, token, gauge, _, amount_mantissa = line.strip().split(",")
+            key = (token, gauge)
+            aggregated_bribes[key] = aggregated_bribes.get(key, 0) + int(
```

Comment:
not sure if python's `int` has enough precision for some of these mantissas, better to use `Decimal`